ChronoVisor/include/Chronicle.h

```diff
@@ -76,94 +84,125 @@ class Chronicle {
 
     friend std::ostream& operator<<(std::ostream& os, const Chronicle& chronicle);
 
-    bool addProperty(const std::string& name, const std::string& value) {
+    int addProperty(const std::string& name, const std::string& value) {
         if (propertyList_.size() <= MAX_CHRONICLE_PROPERTY_LIST_SIZE) {
-            auto res = propertyList_.emplace(name, value);
-            return res.second;
+            auto res = propertyList_.insert_or_assign(name, value);
+            if (res.second) return CL_SUCCESS;
+            else return CL_ERR_UNKNOWN;
         } else {
-            return false;
+            return CL_ERR_CHRONICLE_PROPERTY_FULL;
         }
     }
 
-    bool addMetadata(const std::string& name, const std::string& value) {
+    int addMetadata(const std::string& name, const std::string& value) {
         if (metadataMap_.size() <= MAX_CHRONICLE_METADATA_MAP_SIZE) {
-            auto res = metadataMap_.emplace(name, value);
-            return res.second;
+            auto res = metadataMap_.insert_or_assign(name, value);
+            if (res.second) return CL_SUCCESS;
+            else return CL_ERR_UNKNOWN;
         } else {
-            return false;
+            return CL_ERR_CHRONICLE_METADATA_FULL;
         }
     }
 
-    bool addStory(std::string &chronicle_name, const std::string& story_name,
-                  const std::unordered_map<std::string, std::string>& attrs) {
-        Story *pStory = new Story();
-        pStory->setName(story_name);
-        pStory->setProperty(attrs);
+    int addStory(std::string &chronicle_name, const std::string& story_name,
+                 const std::unordered_map<std::string, std::string>& attrs) {
         // add cid to name before hash to allow same story name across chronicles
         std::string story_name_for_hash = chronicle_name + story_name;
         uint64_t cid = CityHash64(chronicle_name.c_str(), chronicle_name.size());
         uint64_t sid = CityHash64(story_name_for_hash.c_str(), story_name_for_hash.size());
+        if(storyMap_.find(sid) != storyMap_.end()) return CL_ERR_STORY_EXISTS;
+        auto *pStory = new Story();
+        pStory->setName(story_name);
+        pStory->setProperty(attrs);
         pStory->setSid(sid);
         pStory->setCid(cid);
+        std::lock_guard<std::mutex> lock(storyMapMutex_);
```

Comment:
Why do you need this storyMapMutex_? Doesn't every chronicle object have its local story maps? The chronicle map is locked prior to this.